CONTRIBUTING.md

```diff
@@ -71,6 +71,54 @@ Be sure to include any related GitHub issue references in the commit message.  S
 [GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues
 and commits.
 
+### Formatting Code
+
+ We do use Prettier as a tool to ensure a consistent format. A format check step runs as part of our continuous integration and pull requests get failed checks if they don't adhere to the conventions. To make sure your contribution is properly formatted you can run the following:
+
+ ``` shell
+ npm run format-check
+ ```
+
+ The `format` script can also be used to make prettier automatically apply the format guidelines:
+
+ ``` shell
+ npm run format
+ ```
+
+ ### Static Analysis
+
+ As part of our continuous integration workflow we do pass ESlint to all pull requests. To make sure that your contribution passes all the static checks you can use:
+
+ ``` shell
+ npm run lint
+ ```
+
+ ### Tests
+
+ We value highly-tested software. All pull requests should come accompanied by corresponding unit tests. Integration tests are also welcomed. To make sure that your contribution isn't causing any regression we run the test suite as part of our continuous integration process. You should also make sure that all the tests pass before sending your pull request:
+
+ ``` shell
+ npm run test
+ ```
+
+ ## Release Process
+
+ All stable code is hosted at the `main` branch. Release and version management is fully automated and based in commit messages that follow a subset of the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec. Based on the commit message, our continuous integration pipeline will automatically build, package and version bump and tag the codebase. 
+
+ Versioning follows the [semver](https://semver.org) spec and the appropriate patch, minor or major version is bumped depending on the commit messages. Here is the convention used for version bumping based on the commit messages:
+
+ * If the strings "BREAKING CHANGE", "major" are found anywhere in any of the commit messages or descriptions the major version will be incremented.
+ * If a commit message begins with the string "feat" or includes "minor" then the minor version will be increased. This works for most common commit metadata for feature additions: "feat: new API" and "feature: new API".
```

Comment:
Do we want to use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)?

Reply:
The action that I am using here for doing version bumps is based on conventional commits but also [allows to customize all the triggering words](https://github.com/phips28/gh-action-bump-version/blob/master/action.yml#L18).